models/users.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+var Db        = require('../config/database');
+var Session   = require('./sessions');
+var validator = require('validator');
+var mongoose  = require('mongoose');
+
+// Create the schema for a user
+var UserSchema = mongoose.Schema({
+  username      : { type: String, required: true },
+  usernameLower : { type: String, required: true },
+  email         : { type: String, required: true },
+  password      : { type: String, required: true }
+});
+
+// Creates a collection named users in MongoDB
+var UserMongoModel = Db.model('users', UserSchema);
+
+// Takes username, password, email and saves a new user to MongoDB
+function createUser(username, password, email, callback) {
+  var newUser = new UserMongoModel({
+    username      : username,
+    usernameLower : username.toLowerCase(),
+    email         : email,
+    password      : password
+  });
+
+  // Check if newUser.username is already in UserMongoModel
+  UserMongoModel.findOne({ usernameLower : username.toLowerCase() }, function(err, user) {
+    if (user) {
+      callback({
+        code    : 400,
+        message : 'Username already exists'
+      });
+    } else {
+
+      // Check if newUser.email is already in UserMongoModel
+      UserMongoModel.findOne({ email : email.toLowerCase() }, function(err, user) {
+        if (user) {
+          callback({
+            code    : 400,
+            message : 'Email already exists'
+          });
+        } else {
+
+          // Saves user to UserMongoModel
+          newUser.save(function(err, user) {
+            if (err) {
+              // TODO: Error message?
+              callback(err);
+            } else if (!user) {
+              callback({
+                code    : 400,
+                message : 'User creation failed.'
+              });
+            } else {
+
+              // Creates a new session with user._id
+              Session.create(user._id, function(err, session) {
+                if (err) {
+                  // Error message handled by session model
+                  callback(err);
+                  return;
+                }
+                callback(null, session);
+              });
+            } 
+          });
+        }
+      });
+    }
+  });
+}
+
+// Takes a token string and deletes the associated user and session from MongoDB
+function deleteUser(clientToken, callback) {
+  Session.findUser(clientToken, function(err, userId) {
+    if (err) {
+      // Error message handled by session model
+      callback(err);
+    } else {
+
+      // Remove user from UserMongoModel
+      UserMongoModel.findByIdAndRemove(userId.toLowerCase(), function(err, user) {
+        if (err) {
+          // TODO: Error message?
+          callback(err);
+        } else {
+
+          // Removes session from sessions collection
+          Session.destroy(clientToken, function(err) {
```

Comment:
We should have a `Session.destroyAll` method instead to destroy all the user's sessions if they delete their account.